Lib/_pyrepl/readline.py

```diff
@@ -219,16 +219,29 @@ def do(self) -> None:
         # if there are already several lines and the cursor
         # is not on the last one, always insert a new \n.
         text = r.get_unicode()
+
         if "\n" in r.buffer[r.pos :] or (
             r.more_lines is not None and r.more_lines(text)
         ):
+            def _newline_before_pos():
+                before_idx = r.pos - 1
+                while before_idx > 0 and text[before_idx].isspace():
+                    before_idx -= 1
+                return text[before_idx : r.pos].count("\n") > 0
             #
-            # auto-indent the next line like the previous line
-            prevlinestart, indent = _get_previous_line_indent(r.buffer, r.pos)
-            r.insert("\n")
-            if not self.reader.paste_mode and indent:
-                for i in range(prevlinestart, prevlinestart + indent):
-                    r.insert(r.buffer[i])
+            # if there's already a new line before the cursor then
+            # even if the cursor is followed by whitespace, we assume
+            # the user is trying to terminate the block
+            if _newline_before_pos() and text[r.pos:].isspace():
+                self.finish = True
+            else:
+                #
```

Comment:
Just out of curiosity, I see some other places in the file that use these empty comment lines before starting the comment block...Is this a style convention? I haven't really seen it used elsewhere.

Reply:
Looks like this was an original style of pyrepl as adopted from PyPy. I agree with you Savannah that it looks out of place. We probably want to get rid of it.